### PR TITLE
WIP: Unprotect["`*"] and Protect["`*"]

### PR DESCRIPTION
**Pull request by rocky**
"`*" is the same as "*`*" - is this right?

### CODE REVIEW

**Comment by rocky** [comment]
@GarkGarcia and @mmatera I've been trying to get 

```
Import@"https://raw.githubusercontent.com/jkuczm/MathematicaSyntaxAnnotations/master/NoInstall.m"
```

or any of the Noinstall.m's working that are over at repos in https://github.com/jkuczm/

If you have a chance could you look at this PR and the failure in:


```
In[1]:=Import@"https://raw.githubusercontent.com/jkuczm/MathematicaSyntaxAnnotations/master/NoInstall.m"

Clear::wrsym: Symbol $OperatingSystem is Protected.  Clear::wrsym:
Symbol List is Protected.  Clear::wrsym: Symbol Symbol is Protected.
Clear::wrsym: Symbol {SyntaxAnnotations`} is Protected.  Clear::wrsym:
Symbol Unix is Protected.  Clear::wrsym: Symbol List is Protected.
Clear::wrsym: Symbol Symbol is Protected.  $RecursionLimit::reclim:
Recursion depth of 200 exceeded.  Out[0]= $Aborted
```

The sense I am gertting is that the Recursion depth of 200 may mean that one command has an error itereating over all symbols and there are over 200 such errors. This probalby is a separate problem too. 

I won't have time to work on this for a while, so just continue if you can and are able.

File: mathics/builtin/attributes.py [hunk 1]
```diff
@@ -173,6 +191,19 @@ class Unprotect(Builtin):
         'Unprotect[symbols__]': 'ClearAttributes[{symbols}, Protected]',
     }
 
+    def apply(self, pattern, evaluation):
+        "Unprotect[pattern_]"
```
Comment (mmatera) [inline mathics/builtin/attributes.py:195]: The defined behavior is different from the established in the docstring: this does not accept a list of patterns, but just one.  
,

**Review by mmatera — COMMENTED** [review]
Let go by parts, like Jack...
"`*" is equivalent to "Everything in the current context", and should be equivalent to "*"
while "*`*" seems to indicate "Everything in every Context".

**Comment by mmatera** [comment]
Regarding the a..  and a... issue, I found that the \$RecursionLimig  happens when the parser tryies to format Repeated or RepeatedNull. I am working on that, but slowly...

File: mathics/builtin/attributes.py (hunk 1 above)
Comment (rocky) [reply mathics/builtin/attributes.py:195]: Thanks for pointing out. 

Yes, that should probably be `pattern__` and the code will have to get more complicated too. 

I got the docstring from Mathematica. And I doubt the code handles _form_ like it should as well.

However, the particular problem I wanted to fix was `Unprotect["\[GraveAccent]*]` though which explains the discrepancy. 

I'll address this when I have a chance to come back to it.

**Comment by rocky** [comment]
> Regarding the a.. and a... issue, I found that the `$RecursionLimit` happens when the parser tries to format `Repeated` or `RepeatedNull`. I am working on that, but slowly...

It has nothing to do with the parser, and everything to do with the formatter. I have updated issue #974

File: mathics/core/definitions.py [hunk 2]
```diff
@@ -296,10 +296,13 @@ def get_matching_names(self, pattern) -> typing.List[str]:
...
-            ctx_pattern = ((ctx_pattern + '`')
-                           .replace('@', '[^A-Z`]+')
-                           .replace('*', '.*')
-                           .replace('$', r'\$'))
+            if ctx_pattern == "":
+                ctx_pattern="System`"
+            else:
+                ctx_pattern = ((ctx_pattern + '`')
+                               .replace('@', '[^A-Z`]+')
+                            .replace('*', '.*')
```
Comment (GarkGarcia) [inline mathics/core/definitions.py:304]: We should probably align this properly before merging.

**Comment by rocky** [comment]
All of these changes are subsumed in #1003

**Pull request closed without merging**